src/lib.rs

```diff
@@ -1236,6 +1236,58 @@ pub trait Itertools : Iterator {
     {
         self.sorted_by(cmp)
     }
+
+    /// Return the minimum and maximum element of an iterator.
+    ///
+    /// For the minimum, the first minimal element is returned.  For the maximum,
+    /// the last maximal element wins.  This matches the behavior of the standard
+    /// `Iterator::min()` and `Iterator::max()` methods.
+    ///
+    /// This requires `clone()`-able elements since the returned minimum and
+    /// maximum might actually be the same item.
+    fn minmax(mut self) -> Option<(Self::Item, Self::Item)>
+        where Self: Sized, Self::Item: Ord, Self::Item: Clone
```

Comment:
I think std is too conservative, and that this is a typical case where floats will want to be used as well. Ord vs PartialOrd is the user's choice: they choose what type to input. Unfortunately there may not be a nice way to put all Ord types and floats under the same trait, since trait coherence will maintain that f32 may impl Ord in the future, ruling it conflicting.

If it's not possible, then we'll skip the float concerns for now.